src/LanguageServer/Protocol/Features/Diagnostics/DocumentAnalysisExecutor_Helpers.cs

```diff
@@ -146,7 +146,24 @@ static string GetLanguageSpecificId(string? language, string noLanguageId, strin
             // Create driver that holds onto compilation and associated analyzers
             var filteredProjectAnalyzers = projectAnalyzers.WhereAsArray(static a => !a.IsWorkspaceDiagnosticAnalyzer());
             var filteredHostAnalyzers = hostAnalyzers.WhereAsArray(static a => !a.IsWorkspaceDiagnosticAnalyzer());
-            var filteredProjectSuppressors = filteredProjectAnalyzers.WhereAsArray(static a => a is DiagnosticSuppressor);
+            var hostAnalyzersSegmentWithSuppressors = filteredHostAnalyzers.AsMemory()[filteredHostAnalyzers.FindRange(static a => a is DiagnosticSuppressor)];
```

Comment:
Is it guaranteed that suppressors are in a single contiguous range?

Reply:
Oh, never mind, `Equals` will return false on those within the range that aren't suppressors.

Reply:
A clarifying comment would be useful.

Reply:
Why not used pooled hash set for the lookup? Would be simpler.